vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp

```diff
@@ -272,37 +274,163 @@ void ScrollViewComponentView::OnPointerDown(const winrt::Windows::UI::Input::Poi
+      args.Handled(lineRight(false));
+      break;
+  }
+}
 
-  if (ptViewport.x >= 0 && ptViewport.x <= m_layoutMetrics.frame.size.width && ptViewport.y >= 0 &&
-      ptViewport.y <= m_layoutMetrics.frame.size.height) {
-    m_scrollVisual.ScrollBy({0, static_cast<float>(-delta), 0});
-    return true;
+bool ScrollViewComponentView::scrollToEnd(bool animate) noexcept {
+  if ((((m_contentSize.height - m_layoutMetrics.frame.size.height) * m_layoutMetrics.pointScaleFactor) -
+       m_scrollVisual.ScrollPosition().y) < 1.0f) {
+    return false;
   }
 
-  return false;
+  auto x = (m_contentSize.width - m_layoutMetrics.frame.size.width) * m_layoutMetrics.pointScaleFactor;
+  auto y = (m_contentSize.height - m_layoutMetrics.frame.size.height) * m_layoutMetrics.pointScaleFactor;
+  m_scrollVisual.TryUpdatePosition({static_cast<float>(x), static_cast<float>(y), 0.0f}, animate);
+  return true;
+}
+
+bool ScrollViewComponentView::scrollToStart(bool animate) noexcept {
+  m_scrollVisual.TryUpdatePosition({0.0f, 0.0f, 0.0f}, animate);
+  return true;
+}
+
+bool ScrollViewComponentView::pageUp(bool animate) noexcept {
+  if (m_scrollVisual.ScrollPosition().y <= 0.0f) {
+    return false;
+  }
+
+  float scrollToVertical =
+      m_scrollVisual.ScrollPosition().y - (m_layoutMetrics.frame.size.height * m_layoutMetrics.pointScaleFactor);
+  float scrollToHorizontal = m_scrollVisual.ScrollPosition().x;
+  m_scrollVisual.TryUpdatePosition({scrollToHorizontal, scrollToVertical, 0.0f}, false /*animate*/);
+  return true;
+}
+
+bool ScrollViewComponentView::pageDown(bool animate) noexcept {
+  if ((((m_contentSize.height - m_layoutMetrics.frame.size.height) * m_layoutMetrics.pointScaleFactor) -
+       m_scrollVisual.ScrollPosition().y) < 1.0f) {
+    return false;
+  }
+
+  float scrollToVertical =
+      m_scrollVisual.ScrollPosition().y + (m_layoutMetrics.frame.size.height * m_layoutMetrics.pointScaleFactor);
+  float scrollToHorizontal = m_scrollVisual.ScrollPosition().x;
+  m_scrollVisual.TryUpdatePosition({scrollToHorizontal, scrollToVertical, 0.0f}, false /*animate*/);
+  return true;
+}
+
+bool ScrollViewComponentView::lineUp(bool animate) noexcept {
+  return scrollUp(c_scrollerLineDelta * m_layoutMetrics.pointScaleFactor, animate);
+}
+
+bool ScrollViewComponentView::lineDown(bool animate) noexcept {
+  return scrollDown(c_scrollerLineDelta * m_layoutMetrics.pointScaleFactor, animate);
+}
+
+bool ScrollViewComponentView::lineLeft(bool animate) noexcept {
+  if (m_scrollVisual.ScrollPosition().x <= 0.0f) {
+    return false;
+  }
+
+  float scrollToVertical = m_scrollVisual.ScrollPosition().y;
+  float scrollToHorizontal =
+      m_scrollVisual.ScrollPosition().x - (c_scrollerLineDelta * m_layoutMetrics.pointScaleFactor);
+  m_scrollVisual.TryUpdatePosition({scrollToHorizontal, scrollToVertical, 0.0f}, false /*animate*/);
+  return true;
+}
+
+bool ScrollViewComponentView::lineRight(bool animate) noexcept {
```

Comment:
If the scroll view isn't set to be horizontal is this set up to correctly do nothing?

Reply:
Right now it'll scroll horizontally if the content is larger than the viewport.  -- There are a bunch of properties to control scroll enable/disable which are not yet implemented.